commons/src/main/java/org/calrissian/accumulorecipes/commons/transform/KeyToTupleCollectionWholeColFXform.java

```diff
@@ -32,80 +36,69 @@
-        return kryo;
-    }
+      V entry = buildEntryFromKey(keyValueEntry.getKey());
 
+      List<Map.Entry<Key,Value>> groupedKVs = decodeRow(keyValueEntry.getKey(), keyValueEntry.getValue());
 
-    @Override
-    public V apply(Map.Entry<Key, Value> keyValueEntry) {
-        try {
-            Map<Key, Value> keyValues = decodeRow(keyValueEntry.getKey(), keyValueEntry.getValue());
-            V entry = null;
-
-            for (Map.Entry<Key, Value> curEntry : keyValues.entrySet()) {
-
-                if (entry == null)
-                    entry = buildEntryFromKey(curEntry.getKey());
-                String[] colQParts = splitPreserveAllTokens(curEntry.getKey().getColumnQualifier().toString(), NULL_BYTE);
-                String[] aliasValue = splitPreserveAllTokens(colQParts[1], ONE_BYTE);
-                String visibility = curEntry.getKey().getColumnVisibility().toString();
-
-                try {
-                    Collection<Map<String,Object>> meta = metadataSerDe.deserialize(curEntry.getValue().get());
-                    if(meta != null) {
-                        for(Map<String,Object> curMeta : meta) {
-                            Map<String,Object> metadata = new HashMap<String,Object>();
-                            if(curMeta != null)
-                                metadata.putAll(curMeta);
-
-                            setVisibility(metadata, visibility);
-                            Tuple tuple = new Tuple(colQParts[0], typeRegistry.decode(aliasValue[0], aliasValue[1]), metadata);
-                            entry.put(tuple);
-                        }
-                    } else {
-                        Map<String,Object> metadata = new HashMap<String,Object>();
-                        setVisibility(metadata, visibility);
-                        Tuple tuple = new Tuple(colQParts[0], typeRegistry.decode(aliasValue[0], aliasValue[1]), metadata);
-                        entry.put(tuple);
-                    }
-                } catch(Exception e) {
-                    log.error("There was an error deserializing the metadata for a tuple", e);
-                }
-            }
+      for (Map.Entry<Key,Value> groupedEvent : groupedKVs) {
+
+        List<Map.Entry<Key,Value>> keyValues = decodeRow(groupedEvent.getKey(), groupedEvent.getValue());
+
+        for (Map.Entry<Key,Value> curEntry : keyValues) {
 
-            return entry;
-        } catch (IOException e) {
-            throw new RuntimeException(e);
+          String[] colQParts = splitPreserveAllTokens(curEntry.getKey().getColumnQualifier().toString(), NULL_BYTE);
+          String[] aliasValue = splitPreserveAllTokens(colQParts[1], ONE_BYTE);
+          String visibility = curEntry.getKey().getColumnVisibility().toString();
+
+          try {
+            Map<String,Object> meta = metadataSerDe.deserialize(curEntry.getValue().get());
+            if (meta != null) {
+              Map<String,Object> metadata = new HashMap<String,Object>();
```

Comment:
Read comment above about not doing the if else around all of the logic and only around the creation of the metadata.